### DISCUSSION TITLE
/alerts does not include some future events

### DISCUSSION THREAD

**Discussion by potion-cellar** [weather-gov/api#711] [General]
Hello,

We consume NWS alerts using various `/alerts` endpoints and one thing we've noticed is that some alerts as visible on the weather.gov map are impossible to retrieve via `/alerts`, no matter the filters used or trying to query specific zones / wfos.

Today (13:30 MST, Mar 27 2024) there is a Flood Watch in Maine (`MEZ011-015>017-029>032-280515-/O.NEW.KCAR.FA.A.0003.240328T1500Z-240330T0000Z//00000.0.ER.000000T0000Z.000000T0000Z.000000T0000Z.OO/`) and a Winter Weather Advisory in Idaho (`IDZ006-281730-/O.NEW.KMSO.WW.Y.0030.240328T0600Z-240328T1800Z/`) that cannot be retrieved via the API.

All of the missing alerts are effective in the future. However, there are some alerts which have an `onset` time in the future that are properly returned by the API -- they all have an `effective` time in the past. Guessing the `effective` time for the missing alerts above is still in the future.

Is there something I'm missing here? Since this API also feeds into the mapservice that powers https://radar.weather.gov and https://www.wrh.noaa.gov/map/, these maps don't match the map on weather.gov, which seems odd.

**Comment by Todd812** [comment]
I'm also seeing the same thing.  Like currently New Mexico has a Fire Weather Watch as of 5:31 EST but it's not showing on the CAP page at the NWS.  It's also not showing inside the API.  But it is showing up when local alerts are clicked from the weather.gov website.  It's also not working with the future alerts for frost or freeze in Norther Mississippi.  I've updated my correlation file and it's still not matching.

**Comment by akrherz** [comment]
Just issued [admin notice](https://www.nco.ncep.noaa.gov/status/messages/)
```
NOUS42 KWNO 272140
ADASDM

SENIOR DUTY METEOROLOGIST NWS ALERT ADMINISTRATIVE MESSAGE
NWS NCEP CENTRAL OPERATIONS COLLEGE PARK MD
2140Z WED MAR 27 2024


...NGITWS WWA PROCESSING RESTORED...
WWA processing through NGITWS/ALERTS/API has been restored as
of 2052Z.
```

**Reply by Todd812** [reply]
Hello akrherz,

Maybe I'm looking at something wrong.  Here is a current fire alerts in New Mexico.  
https://forecast.weather.gov/showsigwx.php?warnzone=NMZ231&warncounty=NMC021&firewxzone=NMZ104&local_place1=13%20Miles%20NW%20Logan%20NM&product1=Fire+Weather+Watch&lat=35.5083&lon=-103.5868

Now calling on the CAP site shows nothing:  
https://alerts.weather.gov/search?area=NM
https://alerts.weather.gov/search?zone=NMZ232
https://alerts.weather.gov/search?zone=NMZ104

API Shows Nothing:
https://api.weather.gov/alerts/active?area=NM&urgency=Future
https://api.weather.gov/alerts/active?area=NM
https://api.weather.gov/alerts?event=Fire%20Weather%20Watch&area=NM&limit=500
https://api.weather.gov/alerts/active?zone=NMZ104&urgency=Future&limit=500
https://api.weather.gov/alerts/active?zone=NMZ104&urgency=Future&limit=500
https://api.weather.gov/alerts/active?zone=NMZ232

How would you get the correct alerts for the state of New Mexico?  Actually this has been going on for a little while with these alert types mostly fire weather in certain states.  Was something changed.  I looked for a change notice but didn't see anything.

Thanks and have a great day

**Comment by potion-cellar** [comment]
@akrherz @Todd812 It sounds like they might have fixed the issue -- but not sure if they backfilled any of the missing alerts. I'm still not seeing the alerts in OP (in addition to what @Todd812 posted) but these were all issued before the fix was implemented.

**Reply by Todd812** [reply]
Hello Potion-Cellar,

It's not fixed and alerts aren't showing.  I think my email to the API email got ignored pretty much.  But we will see, I'm sure there were plenty more.  I would've posted about this error in the API but commented because you already found it.  Alerts should immediately show up on all platforms and feeds.  So I was alerted about the problem by my server.  So I came here and found your post.  Hopefully it doesn't happen again.  It would seem that only those alerts posed here are working with the API:  https://www.wrh.noaa.gov/map/

Todd

**Reply by potion-cellar** [reply]
@Todd812 That map uses a webservice that pulls alerts from the API (radar.weather.gov does as well) so yeah it's a good source for viewing what alerts are actually present in the API (and comparing to weather.gov map and professional met software that gets alerts from other channels)
